haystack/nodes/prompt/prompt_node.py

```diff
@@ -139,6 +139,13 @@ def prepare(self, *args, **kwargs) -> Dict[str, Any]:
         if kwargs:
             for param in self.prompt_params:
                 if param in kwargs:
+                    if param == "documents":
+                        for doc in kwargs.get("documents", []):
+                            if not isinstance(doc, str) and not isinstance(doc.content, str):
+                                raise ValueError("PromptNode only accepts text documents.")
+                        kwargs["documents"] = [
+                            doc.content if isinstance(doc, Document) else doc for doc in kwargs.get("documents", [])
+                        ]
```

Comment:
This has been moved from `PromptNode.run`to access `Document` objects within Templates.